Условие 1: 1 задача
Скачать файл в уроке
Считать данные с помощью pandas
Вывести на экран первые 5 строк
Посмотреть на описание признаков и на их содержание

Условие 2: 2 задача
Проведите первичный анализ данных
Изучите типы данных
Найдите количество пропущенных ячеек в данных
Посчитайте основные статистики по всем признакам и поизучайте их
Пишите выводы

Условие 3: 3 задача
Ответьте на несколько вопросов
3.1 В каком диапазоне изменяются стоимости недвижимости?
3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
3.3 Как много домов с разными этажами в данных?
3.4 Насколько хорошие состояния у домов в данных?
3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?
Условие 4: 4 задача
Ответьте на несколько вопросов

4.1 Сколько в среднем стоят дома, у которых 2 спальни?
4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
4.3 Как много домов коснулся ремонт?
4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

Условие 5: 5 задача
Ответьте на несколько вопросов
5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

- id - Уникальный ID для каждого дома
- date - Дата продажи дома
- price - Стоимость продажи дома
- bedrooms - Кол-во спален
- bathrooms - Кол-во ванных комнат (0.5 - туалет без душа)
- sqft_living - Кв. метры жилые
- sqft_lot - Кв. метры общие
- floors - Кол-во этажей
- waterfront - Есть набержная или нет
- view - Значение от 0 до 4 насколько хороший вид
- condition - Значение от 1 до 5 насколько хорошее состояние
- grade - Значение от 1 до 13, где 1-3 плохая конструкция здания и дизайн, 7 - средний уровень конструкции и дизайна, 11-13 - высокое качество конструкции и дизайна
- sqft_above - Кв. метры дома, которые находятся выше земли
- sqft_basement - Кв. метры дома, которые находятся ниже земли
- yr_built - Год постройки дома
- yr_renovated - Год ремонта дома
- zipcode - Индекс
- lat - Широтма
- long - Долгота
- sqft_living15 - Кв. метры жилой площади у 15 соседей
- sqft_lot15 - Кв. метры общей площади у 15 соседей


In [88]:
import pandas as pd
df = pd.read_csv('kc_house_data.csv', sep = ',', encoding='latin-1')


Условие 2: 2 задача
Проведите первичный анализ данных
Изучите типы данных
Найдите количество пропущенных ячеек в данных
Посчитайте основные статистики по всем признакам и поизучайте их
Пишите выводы

In [89]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [90]:
df.isnull().sum().sum()

0

In [91]:
df.describe()


id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

Типы данных:

    16 количественных признаков (int64, float64)
    5 качественных признаков (object)

Пропущенные значения:

    Отсутствуют

Средняя цена:

Средняя цена дома в данном наборе данных составляет 4 580 302 000.

Другие выводы:

    Средняя площадь дома: 2079.899736 м²
    Среднее количество комнат: 3.37
    Среднее количество ванных комнат: 2.11
    Большинство домов:
        Построены в 1970-х годах
        Находятся в хорошем состоянии



Условие 3: 3 задача
Ответьте на несколько вопросов
3.1 В каком диапазоне изменяются стоимости недвижимости?
3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
3.3 Как много домов с разными этажами в данных?
3.4 Насколько хорошие состояния у домов в данных?
3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [92]:
df.describe()['price'].loc[['min', 'max']]

min      75000.0
max    7700000.0
Name: price, dtype: float64

In [93]:
df['dolya'] = df['sqft_living'] / df['sqft_lot']
sr_dolya = df['dolya'].mean()
print(f"средняя доля жилой площади: {sr_dolya:.2%}")

средняя доля жилой площади: 32.37%


In [94]:

df['floors'].unique().shape[0]

6

In [95]:
a=df['condition'].mean()
print(f" Среднее состояние домов по оценке от 1 до 5: {a}")

 Среднее состояние домов по оценке от 1 до 5: 3.4094295100171195


In [96]:
df.describe()['yr_built'].loc[['min', 'max']]

min    1900.0
max    2015.0
Name: yr_built, dtype: float64

4.1 Сколько в среднем стоят дома, у которых 2 спальни?
4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
4.3 Как много домов коснулся ремонт?
4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?


In [97]:
s= df[df['bedrooms'] == 2]['price'].mean()
print(f'средняя цена дома у которого две спальни = {s}')

средняя цена дома у которого две спальни = 401372.681884058


In [101]:

df1 = df[df['price'] > 6000000]
average_area = df1['sqft_lot'].mean()
print(average_area)

32099.666666666668


In [99]:
df1 = df[df['yr_renovated'] != 0]

# Подсчет количества отремонтированных домов
renovated_houses = df1.shape[0]

# Вывод
print(f"Количество отремонтированных домов: {renovated_houses}")

Количество отремонтированных домов: 914


4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [100]:
e = df[df['grade'] > 10]["price"].mean()
h = df[df['grade'] < 4]["price"].mean()
print(e-h)


1488885.1175298805


Условие 5: 5 задача
Ответьте на несколько вопросов
5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [111]:
df[(df['waterfront'] == 1) & (df['bathrooms'] >= 3) & (df['sqft_basement'] > 0)].shape[0]

41

In [124]:
a = df[((df['view'] == 4 ) | (df['waterfront'] == 1)) & (df['condition'] ==5) & (df['yr_built'] >= 1980)]
a['price'].min(),a['price'].max()

(1295000.0, 3000000.0)

In [126]:
df[(df['sqft_basement']==0) & (df['floors'] == 2) & (df['price'] < 150000)]['condition'].mean()
5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

2.8333333333333335